### PR TITLE
support workload identity federation

### PR DESCRIPTION
**Pull request by fairclothjm**
This PR adds plugin WIF support to the Azure auth method. The following new fields are added to the config endpoint to enable configuring Workload Identity Federation:

- identity_token_audience
- identity_token_ttl

### CODE REVIEW

File: bootstrap/terraform/vm.tf [hunk 1]
```diff
@@ -111,7 +111,7 @@ resource "azurerm_public_ip" "vault_azure_pub_ip" {
 
 # Restrict SSH access to the local public IP
 data "http" "my_ip" {
-  url = "https://ifconfig.me/ip"
+  url = "https://ipinfo.io/ip"
```
Comment (thyton) [inline bootstrap/terraform/vm.tf:114]: For my learning purpose, could you share we made this change?

Comment (fairclothjm) [reply bootstrap/terraform/vm.tf:114]: I was unable to ssh into the Azure VM. "https://ifconfig.me/ip" returns an IPv6 and switching to "https://ipinfo.io/ip" resolves the issue. It returns IPv4. Maybe the original "https://ifconfig.me/ip" endpoint changed its behavior at some point?

**Review by thyton — APPROVED** [review]
LGTM. Thank you for reorganizing and refactoring the tests!

File: azure.go [hunk 2]
```diff
@@ -330,6 +380,8 @@ func (b *azureAuthBackend) getAzureSettings(ctx context.Context, config *azureCo
 	}
 	settings.ClientSecret = clientSecret
 
+	settings.IdentityTokenAudience = config.IdentityTokenAudience
+
```
Comment (Zlaticanin) [inline azure.go:384]: One small nit, we probably should set the token ttl here as well
```suggestion
settings.IdentityTokenTTL = config.IdentityTokenTTL
```

**Review by Zlaticanin — APPROVED** [review]
I went through the setup and was able to successfully use the WIF flow. Great job! 👏 🎉

File: azure.go (hunk 2 above)
Comment (vinay-gopalan) [reply azure.go:386]: Resolving since this was added in as part of the context removal PR!

**Review by vinay-gopalan — APPROVED** [review]
LGTM!

**Pull request merged**